diff --git a/tests/check-ebcdic b/tests/check-ebcdic
index 1bad48d..3c78d6f 100755
--- a/tests/check-ebcdic
+++ b/tests/check-ebcdic
@@ -19,14 +19,14 @@ if $iconv -l | grep IBM-1047 > /dev/null; then
     chtag -r tmp-ok-x15
     chtag -r tmp-ok-x25
   fi
-
-  printf '\150\145\154\154\157\012' > tmp-ok-lf
-  printf '\150\145\154\154\157\302\205' > tmp-ok-nel
+  printf 'hello\n' > tmp-ok-lf
+  printf 'hello\302\205' > tmp-ok-nel
 
   printf '\210\205\223\223\226\025' > tmp-ok-x15
   printf '\210\205\223\223\226\045' > tmp-ok-x25
 
   # Check that by default, EBCDIC 0x15 maps to U+0085.
+  export ICONV_EBCDIC_ZOS_UNIX=JUNK
   unset ICONV_EBCDIC_ZOS_UNIX
 
   $iconv -f UTF-8 -t IBM-1047 < tmp-ok-nel > tmp-out1
