diff --git a/tests/check-ebcdic b/tests/check-ebcdic
index 62dfd61..12eaeec 100755
--- a/tests/check-ebcdic
+++ b/tests/check-ebcdic
@@ -6,13 +6,28 @@ iconv=../src/iconv_no_i18n
 # This test is only meaningful when the EBCDIC encodings are included.
 if $iconv -l | grep IBM-1047 > /dev/null; then
 
-  printf 'hello\n' > tmp-ok-lf
-  printf 'hello\302\205' > tmp-ok-nel
+  if uname -s | grep 'OS/390' >/dev/null ; then
+    # On z/OS, make all the files 'untagged'
+    rm -f tmp-ok-lf tmp-ok-nel tmp-ok-x15 tmp-ok-x25
+    touch tmp-ok-lf
+    touch tmp-ok-nel
+    touch tmp-ok-x15
+    touch tmp-ok-x25
+    chtag -r tmp-ok-lf
+    chtag -r tmp-ok-nel
+    chtag -r tmp-ok-x15
+    chtag -r tmp-ok-x25
+  fi
+
+  printf '\150\145\154\154\157\012' > tmp-ok-lf
+  printf '\150\145\154\154\157\302\205' > tmp-ok-nel
 
   printf '\210\205\223\223\226\025' > tmp-ok-x15
   printf '\210\205\223\223\226\045' > tmp-ok-x25
 
   # Check that by default, EBCDIC 0x15 maps to U+0085.
+  # export required because /bin/sh on z/OS fails if unsetting a variable not set when 'set -e' specified
+  export ICONV_EBCDIC_ZOS_UNIX=JUNK 
   unset ICONV_EBCDIC_ZOS_UNIX
 
   $iconv -f UTF-8 -t IBM-1047 < tmp-ok-nel > tmp-out1
