diff --git a/src/iconv.c b/src/iconv.c
index ec4caa1..c071ae8 100644
--- a/src/iconv.c
+++ b/src/iconv.c
@@ -43,6 +43,11 @@
 #include "uniwidth.h"
 #include "uniwidth/cjk.h"
 
+#ifdef __MVS__
+#include <_Ccsid.h>
+#include "zos-tag.h"
+#endif
+
 /* Ensure that iconv_no_i18n does not depend on libintl.  */
 #ifdef NO_I18N
 #include <stdarg.h>
@@ -674,7 +679,7 @@ static void conversion_error_other (int errnum, const char* infilename)
 
 /* Convert the input given in infile.  */
 
-static int convert (iconv_t cd, int infile, const char* infilename)
+static int convert (iconv_t cd, int infile, const char* infilename, const char* tocode)
 {
   char inbuf[4096+4096];
   size_t inbufrest = 0;
@@ -835,6 +840,11 @@ static int convert (iconv_t cd, int infile, const char* infilename)
     goto done;
   }
  done:
+#ifdef __MVS__
+  if (!status) {
+    status = tagfile(fileno(stdout), tocode);
+  }
+#endif
   if (outbuf != initial_outbuf)
     free(outbuf);
   return status;
@@ -1113,7 +1123,7 @@ int main (int argc, char* argv[])
     if (i == argc)
       status = convert(cd,fileno(stdin),
                        /* TRANSLATORS: A filename substitute denoting standard input.  */
-                       _("(stdin)"));
+                       _("(stdin)"),tocode);
     else {
       status = 0;
       for (; i < argc; i++) {
@@ -1129,7 +1139,7 @@ int main (int argc, char* argv[])
                 infilename);
           status = 1;
         } else {
-          status |= convert(cd,fileno(infile),infilename);
+          status |= convert(cd,fileno(infile),infilename,tocode);
           fclose(infile);
         }
       }
